internal/stage_4.go

```diff
@@ -28,11 +30,11 @@ func testAPIVersionErrorCase(stageHarness *test_case_harness.TestCaseHarness) er
 	correlationId := getRandomCorrelationId()
 	apiVersion := getInvalidAPIVersion()
 
-	broker := protocol.NewBroker("localhost:9092")
+	broker := kafka_broker.NewBroker("localhost:9092")
```

Comment:
We should call this "client" or "connection" - `NewBroker` is wrong, we aren't constructing a broker here